### PR TITLE
Connect 4 physics

### PR DESCRIPTION
**Pull request by ASecondGuy**
# Description
I added a new minigame.
It is a physics based version of connect 4.

## Type of change
- [X] New feature (non breaking change that adds functionality)

# Checklist
- [X] My code follows the general style guidelines of the project
- [x] I have perfomed a self-review of my code
- [x] I have commented my code, particulary where it is unclear
- [x] My changes generate no new warnings or errors
- [X] The project compiles and runs correctly

### CODE REVIEW

**Comment by Numenter** [comment]
Node names should be in PascalCase. My reasoning is that godot names all standard nodes in PascalCase.

Some name suggestions.
- Connect/Bounds/SlopeLeft
- Connect/SpawnLeft

**Comment by ASecondGuy** [comment]
@Numenter I ended up rewriting almost everything but you gave me a lot of inspiration.

File: game/games/asecondguy_connect/connect.gd [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	chip.global_position = _spawners[player_id].global_position
+	if chip.connect("sleeping_state_changed", self, "_on_chip_sleep", [chip]) != OK:
+		GameManager.end_game("A fatal error occured.")
+	_chips.call_deferred("add_child", chip)
+
+
+# checks the chip and updates _end_condition
+func _check_game_end_from_chip(at: Vector2, player_id):
+	# check all 3 valid axis
+	for axis in [Vector2(1, 0), Vector2(0, 1), Vector2(1, 1)]:
```
Comment (Numenter) [inline game/games/asecondguy_connect/connect.gd]:
```suggestion
	# check all 4 valid axis
	for axis in [Vector2(1, 0), Vector2(0, 1), Vector2(1, 1), Vector2(-1, 1)]:
```
image

Comment (ASecondGuy) [reply game/games/asecondguy_connect/connect.gd]: How did I forget this.

**Comment by Numenter** [comment]
Game speed is to slow. 
- It takes 4 sec to get the chip, that is way to long.
- The more FPS you got the slower moves the AI. (60fps = ca 4 sec, 400fps = ca. 32 sec. )[to lift the chip up]

**Comment by Numenter** [comment]
Small thing that struck me.

It takes to long to get the chip (ca. 4 sec).

AI behaver
 - The AI moves the chip out of the frame. Yes the player can do that to, but I found that odd.
 - The AI should hover and drop the chip, the drop with momentum is odd.

The chip has to stop moving before the AI picks up the chip? I had one run where the left AI took longer to pick up the chips.

**Comment by ASecondGuy** [comment]
I like the momentum thing because that is kinda in line with the physics part. 
I didn't think the other stuff was that bad but I guess I gotta fix that.

**Pull request merged**